src/main/org/apache/tools/ant/util/ResourceUtils.java

```diff
@@ -666,6 +666,14 @@ private static void copyWithFilterSets(final Resource source, final Resource des
                                            final String outputEncoding,
                                            final Project project)
         throws IOException {
+        if (source instanceof FileResource && dest instanceof FileResource) {
```

Comment:
I think it would be better to use `FileProvider` rather than `FileResource` here (and in the next blocks). Actually I'd probably move the `instanceof` checks into `areSame` (and replace them with `resource.as(FileProvider.class)` there. This allows for other resources that know they are working on files and removes at least one line of duplicate code per `if` block :-)

Reply:
That makes sense. I have changed it to follow your suggestion and updated this PR. I am going to run this against both windows on linux, through a Jenkins job, before merging this.